File: 2022.1/Programação Orientada a Objetos/lista1java/Ex5.java
public class Ex5{
    public static void main(String args[]) throws Exception{
        int x = EntradaTeclado.leInt();

        int div = 0;
        int minDiv = 99999999;

        for(int i = 2; i <= x; i++){
            if(x%i == 0){
                div++;
                minDiv = Math.min(minDiv,i);
            }
        }

        if(div == 1){
            System.out.println("O número " + x + " é primo");
        }
        else{
            System.out.println("O número " + x + " não é primo");
            System.out.println("Seu menor divisor é o " + minDiv);
        }
    }
}

File: 2022.1/Programação Orientada a Objetos/lista1python/Ex5.py
x = int(input())

div = 0
minDiv = 99999999

for i in range(2,x+1):
    if x%i == 0:
        div += 1
        minDiv = min(minDiv,i)

if div == 1:
    print("O número ", x, " é primo")
else:
    print("O número ", x, " não é primo")
    print("Seu menor divisor é o ", minDiv)